nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/test/java/org/apache/nifi/json/TestJsonTreeRowRecordReader.java

```diff
@@ -296,11 +298,34 @@ void testReadOneLinePerJSON() throws IOException, MalformedRecordException {
 
     @Test
     void testReadMultilineJSON() throws IOException, MalformedRecordException {
+        testReadAccountJson("src/test/resources/json/bank-account-multiline.json", false, null);
+    }
+
+    @Test
+    void testReadJSONStringTooLong() {
+        final StreamConstraintsException mre = assertThrows(StreamConstraintsException.class, () ->
+                testReadAccountJson("src/test/resources/json/bank-account-multiline.json", false, StreamReadConstraints.builder().maxStringLength(2).build()));
+        assertEquals("String length (8) exceeds the maximum length (2)", mre.getMessage());
```

Comment:
Having an assertion for the entire message can make the test brittle. Recommend adjusting it to look for a keyword, in this case `2` to align with the maximum string length.